py/server/tests/test_table.py

```diff
@@ -1115,6 +1115,68 @@ def test_arg_validation(self):
             t.partition_by("A", "B")
         self.assertIn("drop_keys must be", str(cm.exception))
 
+    def test_diff(self):
+        with self.subTest("diff"):
+            t1 = empty_table(10).update(["A = i", "B = i", "C = i"])
+            t2 = empty_table(10).update(["A = i", "B = i % 2 == 0? i: i + 1", "C = i % 2 == 0? i + 1: i"])
+            d = t1.diff(t2, max_lines=10).split("\n")
+            self.assertEqual(len(d), 3)
+            self.assertIn("row 1", d[0])
+            self.assertIn("row 0", d[1])
+
+            d = t1.diff(t2).split("\n")
+            self.assertEqual(len(d), 2)
+
+        with self.subTest("diff - ignore column order"):
+            t1 = empty_table(10).update(["A = i", "B = i + 1"])
+            t2 = empty_table(10).update(["B = i + 1", "A = i"])
+            d = t1.diff(t2, max_lines=10).split("\n")
+            self.assertEqual(len(d), 3)
+
+            t1 = empty_table(10).update(["A = i", "B = i"])
+            t2 = empty_table(10).update(["B = i", "A = i"])
+            d = t1.diff(t2, max_lines=10, ignore_column_order=True)
+            self.assertEqual(len(d), 0)
+
+        with self.subTest("diff - floating_inexact-double"):
+            t1 = empty_table(10).update(["A = i", "B = i + 1.0"])
+            t2 = empty_table(10).update(["A = i", "B = i + 1.00001"])
+            d = t1.diff(t2, max_lines=10, floating_inexact=False).split("\n")
+            self.assertEqual(len(d), 2)
+
+            t1 = empty_table(10).update(["A = i", "B = i + 1.0"])
+            t2 = empty_table(10).update(["A = i", "B = i + 1.00001"])
+            d = t1.diff(t2, max_lines=10, floating_inexact=True).split("\n")
+            self.assertEqual(len(d), 1)
```

Comment:
These line number assertions are very hard for me to assess.  For example, why is this one 1 and not 0?

Reply:
0 means no differences.

Reply:
It is 1 because if you split an empty string with any delimiter, you'll end up with string array of size 1.  Maybe we should return None in place of empty string?

Reply:
That might be a good idea.  It would help avoid accidental weirdness like this.

Reply:
I have decided to keep the return as is. The test cases were updated to make more sense than the awkward way it checks the differences.